app/src/main/java/net/squanchy/notification/NotificationsIntentService.java

```diff
@@ -45,23 +49,33 @@ public void onCreate() {
     }
 
     @Override
+    public void onStart(@Nullable Intent intent, int startId) {
+        super.onStart(intent, startId);
+        subscriptions = new CompositeDisposable();
+    }
+
+    @Override
     protected void onHandleIntent(Intent intent) {
         LocalDateTime notificationIntervalEnd = new LocalDateTime().plusMinutes(NOTIFICATION_INTERVAL_MINUTES);
 
         Observable<List<Event>> sortedFavourites = service.sortedFavourites();
 
         LocalDateTime now = new LocalDateTime();
         if (shouldShowNotifications()) {
-            sortedFavourites
-                    .map(events -> filter(events, event -> event.getStartTime().isAfter(now)))
-                    .map(events -> filter(events, event -> isBeforeOrEqualTo(event.getStartTime(), notificationIntervalEnd)))
-                    .map(notificationCreator::createFrom)
-                    .subscribe(notifier::showNotifications);
+            subscriptions.add(
```

Comment:
I wouldn't clear indiscriminately as every subscription is a separate notification and there might be multiple ones at the same time being prepared (e.g., multiple sessions in the same timeslot). Do they auto-unsubscribe when they complete?